### DISCUSSION TITLE
Connect without username or password - using an Oracle wallet

### DISCUSSION THREAD

**Discussion by daviesluke** [sijms/go-ora#59] [Ideas]
Do you know if it is possible to amend this code to connect to an Oracle DB using a wallet for passwordless connections?

**Comment by sijms** [comment]
sorry for late reply
I didn't try it before and I think it is not supported at this time

now i am working on upgrading the client to high version and i will try to
add this feature

**Comment by sijms** [comment]
auto-login (non-local) oracle wallet is now supported

**Reply by daviesluke** [reply]
That's great - thanks a lot

**Reply by gyanpatel** [reply]
Do you have an example ? 
1. Create wallet 
2. Add creds
3. include the cred in go_ora.Build.Url as connect string 
4. connect to Oracle database 
I tried the same , it works in sqlplus but not via go-ora 

mkstore -wrl %ORACLE_HOME%\network\admin –create   --- creates wallet 

mkstore -wrl %ORACLE_HOME%\network\admin -createCredential connectStr user pass -- adds creds

sqlplus works without user and password, but it does not work from go-ora , it gives invalid user and password, works only when supplied user name and pass. 

sqlplus /@connectStr 

SQL*Plus: Release 19.0.0.0.0 - Production on Fri Apr 4 10:11:16 2025
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.


Connected to:
Oracle Database 11g Enterprise Edition Release 11.2.0.4.0 - 64bit Production
With the Partitioning, Real Application Clusters, Automatic Storage Management, OLAP,
Data Mining and Real Application Testing options

SQL>

**Reply by sijms** [reply]
to use wallet you should pass service name and username
```golang
urlOptions := map[string]string{
	"trace file": "trace.log",
	"wallet":     "./examples/wallet_11/wallet", // path to folder that contain wallet files
}
connStr := go_ora.BuildUrl("localhost", 1521, "service", "user", "", urlOptions)
conn, err := sql.Open("oracle", connStr)
if err != nil {
	log.Fatalln("cannot generate connection: ", err)
}
err = conn.Ping()
if err != nil {
	log.Fatalln("cannot ping: ", err)
}
```

**Reply by gyanpatel** [reply]
sqlplus /@10.185.246.85:1529/**** --> Notice n username and password needed. 

SQL*Plus: Release 19.0.0.0.0 - Production on Sun Apr 6 19:57:47 2025
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.


Connected to:
Oracle Database 11g Enterprise Edition Release 11.2.0.4.0 - 64bit Production
With the Partitioning, Real Application Clusters, Automatic Storage Management, OLAP,
Data Mining and Real Application Testing options

SQL> 

######## 
go -ora --> fails 

2025/04/06 20:01:09 oracle://ops$brdb:@localhost:1521/****?trace file=trace.log&wallet=C%3A%2Fapp%2Fclient%2FSYSTEM%2Fproduct%2F19.0.0%2Fclient_1%2Fnetwork%2Fadmin
2025/04/06 20:01:09 **cannot find credentials for** server: localhost:1521, service: ****,  username: ***


code  :
func setupDbCon() (db *sql.DB) {
	urlOptions := map[string]string{
		"trace file": "trace.log",
		"wallet":     "C:/app/client/SYSTEM/product/19.0.0/client_1/network/admin", // path to folder that contain wallet files
	}
	conStr := go_ora.BuildUrl("localhost", 1521, "****", "****", "", urlOptions)
	log.Println(conStr)
	db, err := sql.Open("oracle", conStr)
	if err != nil {
		log.Fatalln("error connecting to the db  ", err)
		return
	}
	err = db.Ping()
	if err != nil {
		log.Fatalln("error connecting to the db ", err)
		return
	}
	db.SetMaxIdleConns(1)
	log.Println("connected to database successfully using ")
	return
}

what connect string when added to wallet ,  works for you ?

**Reply by gyanpatel** [reply]
It works now, thanks for your help.
